src/data/actions/productSourceOptions.js

```diff
@@ -0,0 +1,37 @@
+import {
+  REQUEST_PRODUCT_SOURCE_OPTIONS,
+  REQUEST_PRODUCT_SOURCE_OPTIONS_FAIL,
+  REQUEST_PRODUCT_SOURCE_OPTIONS_SUCCESS,
+} from '../constants/productSourceOptions';
+
+import DiscoveryDataApiService from '../services/DiscoveryDataApiService';
+import { getErrorMessages } from '../../utils';
+
+export function requestProductSourceOptions(data) {
+  return { type: REQUEST_PRODUCT_SOURCE_OPTIONS, data };
+}
+
+export function requestProductSourceOptionsSuccess(data) {
+  return { type: REQUEST_PRODUCT_SOURCE_OPTIONS_SUCCESS, data };
+}
+
+export function requestProductSourceOptionsFail(error) {
+  return { type: REQUEST_PRODUCT_SOURCE_OPTIONS_FAIL, error };
+}
```

Comment:
We can directly use them inside `fetchProductSourcesOptions` instead of making functions as there is no reusability.